### PR TITLE
Test cases for payloadfactory improvement

### PR DESCRIPTION
**Pull request by keerthu**

### CODE REVIEW

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+import org.testng.annotations.Test;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+
+import static org.testng.Assert.assertFalse;
+
+public class FormatPayloadWithOMTypeCTXExpressionTestCase extends ESBIntegrationTest {
+	@BeforeClass(alwaysRun = true)
+	public void uploadSynapseConfig() throws Exception {
+		super.init();
+		loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/payload/factory/om_ctx_payload_factory_synapse.xml");
```
Comment (bsenduran) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java]: Better to use [File.separator](http://docs.oracle.com/javase/7/docs/api/java/io/File.html#separator) so the test can be run in other OS(windows)

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	public void transformPayloadByCTXPropertyValue() throws AxisFault {
+		sendRobust(getMainSequenceURL(), "IBM");
+	}
+
+
+	private void sendRobust(String trpUrl, String symbol)
+			throws AxisFault {
+		ServiceClient sender;
+		Options options;
+
```
Comment (malakasilva) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java]: Unwanted spacing

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		value2.addChild(fac.createOMText(value1, symbol));
+		value1.addChild(value2);
+		method.addChild(value1);
+
+		return method;
+	}
+
+	@AfterClass(alwaysRun = true)
+	private void destroy() throws Exception {
+		super.cleanup();
```
Comment (malakasilva) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java]: Formatting issue?

**Comment by malakasilva** [comment]
Formatting issues fixed with PR https://github.com/wso2/wso2-synapse/pull/363/files

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+			throws AxisFault {
+		ServiceClient sender;
+		Options options;
+
+		sender = new ServiceClient();
+		options = new Options();
+		options.setProperty(org.apache.axis2.transport.http.HTTPConstants.CHUNKED, Boolean.FALSE);
+		options.setAction("urn:placeOrder");
+
+		if (trpUrl != null && !"null".equals(trpUrl)) {
```
Comment (prabathabey) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java]: I believe, this if clause has to be revisited.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+	public void incomingJsontransformXmlPayloadByJsonTypeProperties() throws Exception {
+
+		loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/payload/factory/xmlFormat_JsonExpressions_CTX.xml");
+		postRequestWithJsonPayload(JSON_PAYLOAD,JSON_TYPE);
+		assertTrue(responsePayload.contains("wso2"), "Symbol wso2 not found in response message");
+	}
+
+	private void postRequestWithJsonPayload(String payload,String contentType) throws Exception{
+
+
```
Comment (prabathabey) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java]: Unnecessary new lines.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+	@SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+	@Test(groups = {"wso2.esb"}, description = "With Stream B&F, xml format, json value, incoming json, outgoing xml ", enabled = false)
+	public void incomingJsontransformXmlPayloadByJsonTypeProperties() throws Exception {
+
+		loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/payload/factory/xmlFormat_JsonExpressions_CTX.xml");
+		postRequestWithJsonPayload(JSON_PAYLOAD,JSON_TYPE);
+		assertTrue(responsePayload.contains("wso2"), "Symbol wso2 not found in response message");
+	}
+
+	private void postRequestWithJsonPayload(String payload,String contentType) throws Exception{
```
Comment (prabathabey) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java]: Formatting issues.

Comment (keerthu) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java]: Formatting issues fixed in https://github.com/wso2/product-esb/pull/363/

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java (hunk 1 above)
Comment (nuwand) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java]: This test will fail on Windows. You must use File.separator.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java [hunk 7]
```diff
@@ -0,0 +1,88 @@
+/*
+*Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (nuwand) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java]: License dates are wrong. Check whether you IDE settings are correct.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+import java.io.Reader;
+import java.io.StringReader;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.net.URL;
+
+import static org.testng.Assert.assertTrue;
+
+public class JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase extends ESBIntegrationTest{
```
Comment (dinusha-dilrukshi) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java]: Better rename class name with camel case. We usually don't use _ in class/method names.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_XmlCTXValue_WithOutStream_TestCase.java [hunk 9]
```diff
@@ -0,0 +1,92 @@
+/*
+*Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (dinusha-dilrukshi) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_XmlCTXValue_WithOutStream_TestCase.java]: Update licence .

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+	}
+
+	@SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+	@AfterClass(alwaysRun = true)
+	private void destroy() throws Exception {
+		try {
+			cleanup();
+		} finally {
+			Thread.sleep(3000);
+//            serverManager.restoreToLastConfiguration();
```
Comment (dinusha-dilrukshi) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java]: Don't keep commented codes.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingXml_XmlCTXValues_WithOutStream_TestCase.java [hunk 11]
```diff
@@ -0,0 +1,75 @@
+/*
+*Copyright (c) 2005-2010, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (dinusha-dilrukshi) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingXml_XmlCTXValues_WithOutStream_TestCase.java]: Update licence.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingXml_XmlCTXValues_WithOutStream_TestCase.java [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+import java.net.URL;
+
+import static org.testng.Assert.assertTrue;
+
+public class JsonFormat_IncomingXml_XmlCTXValues_WithOutStream_TestCase extends ESBIntegrationTest {
+
+	@BeforeClass(alwaysRun = true)
+	public void uploadSynapseConfig() throws Exception {
+		super.init();
+		loadESBConfigurationFromClasspath("/artifacts/ESB/mediatorconfig/payload/factory/jsonFormat_XmlExpressions_CTX.xml");
```
Comment (dinusha-dilrukshi) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingXml_XmlCTXValues_WithOutStream_TestCase.java]: Same as previous test case. Use File.separator.

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java (hunk 7 above)
Comment (keerthu) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/FormatPayloadWithOMTypeCTXExpressionTestCase.java]: This is fixed in https://github.com/wso2/product-esb/pull/363/

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_XmlCTXValue_WithOutStream_TestCase.java (hunk 9 above)
Comment (keerthu) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_XmlCTXValue_WithOutStream_TestCase.java]: This is fixed in https://github.com/wso2/product-esb/pull/363/

File: modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java (hunk 10 above)
Comment (keerthu) [inline modules/integration/tests-integration/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/payload/factory/JsonFormat_IncomingJson_CTXJsonValues_WithStream_TestCase.java]: This is fixed in https://github.com/wso2/product-esb/pull/363/

**Pull request merged**